ImageSegmentation/app/src/main/java/org/pytorch/imagesegmentation/MainActivity.java

```diff
@@ -0,0 +1,166 @@
+
+    public static String assetFilePath(Context context, String assetName) throws IOException {
+        File file = new File(context.getFilesDir(), assetName);
+        if (file.exists() && file.length() > 0) {
+            return file.getAbsolutePath();
+        }
+
+        try (InputStream is = context.getAssets().open(assetName)) {
+            try (OutputStream os = new FileOutputStream(file)) {
+                byte[] buffer = new byte[4 * 1024];
+                int read;
+                while ((read = is.read(buffer)) != -1) {
+                    os.write(buffer, 0, read);
+                }
+                os.flush();
+            }
+            return file.getAbsolutePath();
+        }
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        try {
+            bitmap = BitmapFactory.decodeStream(getAssets().open(imagename));
+        } catch (IOException e) {
+            Log.e("ImageSegmentation", "Error reading assets", e);
+            finish();
+        }
+
+        imageView = findViewById(R.id.imageView);
+        imageView.setImageBitmap(bitmap);
+
+        final Button buttonRestart = findViewById(R.id.restartButton);
+        buttonRestart.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                if (imagename == "deeplab.jpg")
+                    imagename = "dog.jpg";
+                else
+                    imagename = "deeplab.jpg";
+                try {
+                    bitmap = BitmapFactory.decodeStream(getAssets().open(imagename));
+                    imageView.setImageBitmap(bitmap);
+                } catch (IOException e) {
+                    Log.e("ImageSegmentation", "Error reading assets", e);
+                    finish();
+                }
+            }
+        });
+
+
+        buttonSegment = findViewById(R.id.segmentButton);
+        progressBar = (ProgressBar) findViewById(R.id.progressBar);
+        buttonSegment.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                buttonSegment.setEnabled(false);
+                progressBar.setVisibility(ProgressBar.VISIBLE);
+                buttonSegment.setText(getString(R.string.run_model));
+
+                Thread thread = new Thread(MainActivity.this);
+                thread.start();
+            }
+        });
+
+        try {
+            module = Module.load(MainActivity.assetFilePath(getApplicationContext(), "deeplabv3_scripted.pt"));
+        } catch (IOException e) {
+            Log.e("ImageSegmentation", "Error reading assets", e);
+            finish();
+        }
+
+    }
+
+    @Override
+    public void run() {
+        final Tensor inputTensor = TensorImageUtils.bitmapToFloat32Tensor(bitmap,
+                TensorImageUtils.TORCHVISION_NORM_MEAN_RGB, TensorImageUtils.TORCHVISION_NORM_STD_RGB);
```

Comment:
does that model uses completely the same normalization constants as for torchvision pretrained models?

Reply:
yes